bionic/gcs.py

```diff
@@ -2,30 +2,28 @@
 Utilities for working with Google Cloud Storage.
 """
 
-import subprocess
+import logging
 import warnings
 
 from .deps.optdep import import_optional_dependency
-from .utils.urls import bucket_and_object_names_from_gs_url
-
-import logging
 
 logger = logging.getLogger(__name__)
 
-_cached_gcs_client = None
 
+_cached_gcs_fs = None
 
-def get_gcs_client_without_warnings(cache_value=True):
+
+def get_gcs_fs_without_warnings(cache_value=True):
     # TODO This caching saves a lot of time, especially in tests.  But it would
     # be better if Bionic were able to re-use its in-memory cache when creating
     # new flows, instead of resetting the cache each time.
     if cache_value:
```

Comment:
oh I should have checked this before, but no, it doesn't seem expensive. I can remove the cached client, but it would break unit tests. I can take a look at changing unit tests, but do you mind if I leave a TODO and do it in a separate PR? This PR is big as it is.